app/routes.py

```diff
@@ -1 +1,298 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, jsonify, request, make_response, abort
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime 
+# from slack_sdk import WebClient
+import requests
+import os
+
+
+task_bp = Blueprint("task", __name__,url_prefix = "/tasks")
+
+# wave 1 
+# create a task
+@task_bp.route("", methods = ["POST"])
+def create_task():
+    response_body = request.get_json()
+    if "title" not in response_body or "description" not in response_body:
+        return jsonify({ "details": "Invalid data"}), 400
+
+    # using class method
+    new_task = Task.from_dict(response_body)
+
+    db.session.add(new_task)
+    db.session.commit() 
+
+    return jsonify({"task":{
+            "id": new_task.task_id,
+            "title": new_task.title,
+            "description": new_task.description,
+            "is_complete": False
+    }
+    }), 201
+
+
+# get all tasks
+# @task_bp.route("", methods = ["GET"])
+# def get_all_tasks():
+#     response = []
+#     tasks = Task.query.all()
+#     for task in tasks:
+#         response.append(task.to_dict())
+#     return jsonify(response), 200
+
+
+# get one task
+@task_bp.route("/<int:task_id>", methods = ["GET"])
+def get_one_task(task_id):
+    tasks = validate_task(Task,task_id)
+    if tasks:
+        if getattr(tasks, 'goal_id'):
+            response_body = {"task":
+                {"id": tasks.task_id,
+                "goal_id":tasks.goal_id,
+                "title":tasks.title,
+                "description":tasks.description,
+                "is_complete":False
+                }
+                }
+            return jsonify(response_body), 200
+
+    return jsonify({"task":tasks.to_dict()}),200
+
+
+# update task
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    tasks = validate_task(Task,task_id)
+    request_data = request.get_json()
+    
+    tasks.title = request_data["title"]
+    tasks.description = request_data["description"]
+    tasks.completed_at = request_data.get("completed_at", None)
+
+    db.session.commit()
+
+    return jsonify({"task":{
+            "id": tasks.task_id,
+            "title": tasks.title,
+            "description": tasks.description,
+            "is_complete": False
+            }
+            }), 200
+
+
+# delete task
+@task_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    tasks = validate_task(Task,task_id)
+
+    db.session.delete(tasks)
+    db.session.commit()
+
+    return {"details": f'Task {task_id} "Go on my daily walk 🏞" successfully deleted'}
```

Comment:
This message should not be hard-coded like this. Our tests pass, but now every time you delete a task, this message shows up regardless of what the task title actually was. Instead, we should have something like `f"Task {task_id} \"{task.title}\" successfully deleted"`